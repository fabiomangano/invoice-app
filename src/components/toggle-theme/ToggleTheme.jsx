import React from 'react';
import PropTypes from 'prop-types';
import { ReactComponent as LightModeIcon } from '../../assets/icon-moon.svg';
import { ReactComponent as DarkModeIcon } from '../../assets/icon-sun.svg';
import './ToggleTheme.style.scss';

const ToggleTheme = ({ theme, setTheme }) => (
  <button className="toggle-theme" type="button" onClick={setTheme}>
    {theme === 'light' ? <DarkModeIcon /> : <LightModeIcon />}
  </button>
);

ToggleTheme.propTypes = {
  theme: PropTypes.oneOf(['light', 'dark']).isRequired,
  setTheme: PropTypes.func.isRequired,
};

export default ToggleTheme;
