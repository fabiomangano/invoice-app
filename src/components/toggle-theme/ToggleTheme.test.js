import React from 'react';
import Renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ToggleTheme from './ToggleTheme';

describe('ToggleTheme', () => {
  let tree;
  let theme;
  let toggleTheme;

  beforeAll(() => {
    toggleTheme = jest.fn();
    theme = 'light';
  });

  afterEach(() => {
    tree = null;
  });

  it('should render correctly the theme button', () => {
    Renderer.act(() => {
      tree = Renderer.create(
        <ToggleTheme theme={theme} toggleTheme={toggleTheme} />,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should call toggleTheme callback', () => {
    const wrapper = mount(
      <ToggleTheme theme={theme} toggleTheme={toggleTheme} />,
    );
    wrapper.find('button').simulate('click');
    expect(toggleTheme).toHaveBeenCalledTimes(1);
  });
});
