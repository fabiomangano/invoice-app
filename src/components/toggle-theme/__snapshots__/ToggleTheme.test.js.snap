// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleTheme should render correctly the theme button 1`] = `
<button
  className="toggle-theme"
  onClick={[MockFunction]}
  type="button"
>
  <IconMock />
</button>
`;
